### PR TITLE
Video description cleansing/formatting like youtube-dl does

### PR DESCRIPTION
**Pull request by raltamirano**
Turning ```<br />```'s into newlines and unescaping character entities from the (HTML) description of videos.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "video",
     "download"
   ],
-  "version": "0.5.1",
+  "version": "0.5.2",
```
Comment (fent) [inline package.json]: would rather update this myself as I use the command `npm version` that does it

File: lib/util.js [hunk 2]
```diff
@@ -222,3 +222,22 @@ exports.parseFormats = function(info, debug) {
...
+ */
+exports.cleanVideoDescription = function(videoDescriptionHtml) {
+	if (!videoDescriptionHtml || videoDescriptionHtml == null) 
+		return null;
+
+	return entities.decode(videoDescriptionHtml
+		.replace('\n', ' ', 'gi')
+		.replace(/\s*\<\s*br\s*\/?\s*>\s*/gi, '\n')
+		.replace(/<\s*\/\s*p\s*>\s*<\s*p[^>]*>/gi, '\n')
+		.replace(/<.*?>/gi, '')
```
Comment (fent) [inline lib/util.js]: why not use `<[^>]*>` like in the above regex?

File: lib/util.js [hunk 3]
```diff
@@ -222,3 +222,22 @@ exports.parseFormats = function(info, debug) {
...
+/**
+ * @param {String} videoDescriptionHtml
+ * @return {String}
+ */
+exports.cleanVideoDescription = function(videoDescriptionHtml) {
+	if (!videoDescriptionHtml || videoDescriptionHtml == null) 
+		return null;
+
+	return entities.decode(videoDescriptionHtml
+		.replace('\n', ' ', 'gi')
```
Comment (fent) [inline lib/util.js]: says [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) flags in this function is deprecated.

File: test/util-test.js [hunk 4]
```diff
@@ -229,3 +229,17 @@ describe('util.parseFormats()', function() {
...
+describe('util.cleanVideoDescription()', function() {
+  it('Clean the HTML for the video description', function() {
+    var data = require('./files/util/multiline-video-description.json');
+    var cleanDescription = util.cleanVideoDescription(data.html);
+    assert.ok(cleanDescription);
+    assert.equal(cleanDescription, "Some Title\n" +
+      "Line 1\n" +
+      "\"Line 2\"\n" +
+      "1  First Song  5:30\n" +
+      "2  Second Song  5:42");
```
Comment (fent) [inline test/util-test.js]: Can you use single quotes for this?

**Comment by fent** [comment]
This is nice, thanks. Added some comments.

File: lib/util.js (hunk 3 above)
Comment (raltamirano) [inline lib/util.js]: Replaced with:

.replace('/\n/', ' ')

File: test/util-test.js (hunk 4 above)
Comment (raltamirano) [inline test/util-test.js]: Sure.

File: lib/util.js (hunk 2 above)
Comment (raltamirano) [inline lib/util.js]: I can't fully understand these regexes, I've just blindly ported the idea from the "clean_html" in https://github.com/rg3/youtube-dl/blob/master/youtube_dl/utils.py - I'll need some help with this one.

File: lib/util.js (hunk 3 above)
Comment (fent) [inline lib/util.js]: `.replace(/\n/g, ' ')`

File: lib/util.js (hunk 2 above)
Comment (raltamirano) [inline lib/util.js]: After examining video descriptions lines like:

```2  Song name  <a href="#" onclick="yt.www.watch.player.seekTo(5*60+42);return false;">5:42</a><br>```

I think that the reasoning after those two differentes regexes is:

1. First, replace 'br' and an open + closing 'p' html tags with newlines (like the one at the end of the example line)
2. Remove ANY other html tag that remains after the previous step.

I think it works pretty well. Where do you exactly see ```<[^>]*>``` fit in those replacements? Let me know and I'll try to research some more on that direction.

File: lib/util.js [hunk 5]
```diff
@@ -222,3 +222,22 @@ exports.parseFormats = function(info, debug) {
...
+exports.cleanVideoDescription = function(videoDescriptionHtml) {
+	if (!videoDescriptionHtml || videoDescriptionHtml == null) 
+		return null;
+
+	return entities.decode(videoDescriptionHtml
+		.replace(/\n/g, ' ')
+		.replace(/\s*\<\s*br\s*\/?\s*>\s*/gi, '\n')
+		.replace(/<\s*\/\s*p\s*>\s*<\s*p[^>]*>/gi, '\n')
+		.replace(/<.*?>/gi, '')
+	).trim();
```
Comment (fent) [inline lib/util.js]: Can you change these tabs to 2 spaces instead for consistency with the rest of the project?

Comment (raltamirano) [inline lib/util.js]: Sure. Done!

**Comment by fent** [comment]
Can you fix the conflicts? Also, there's another PR that uses the "entities" package instead, what do you think?

**Comment by fent** [comment]
Thanks!

**Comment by raltamirano** [comment]
Glad to help, thank *you* very much for this great tool! I'm using it as the guts for [ytdl-splitter-core] (https://github.com/raltamirano/ytdl-splitter-core), so I'll try my best to keep on collaborating as time allows.

**Pull request merged**